src/app.ts

```diff
@@ -13,14 +13,40 @@ interface TemperatureSummary {
   average: number
 }
 
+let reading: TemperatureReading[];
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  reading = readings;
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
-}
+  let totalTemp: number = 0;
+  let counter: number = 0;
+  let founded: boolean = false;
+  const obj: TemperatureSummary = {
+    first: 0,
+    last: 0,
+    high: 0,
+    low: 0,
+    average: 0
+  };
+
+  reading.forEach(r => {
+    if (r.time.getDate() === date.getDate() && r.city === city) {
+      founded = true;
+      obj.first = !counter ? r.temperature : obj.first;
+      obj.last = r.temperature;
+      obj.high = (r.temperature > obj.high) ? r.temperature : obj.high;
+      obj.low = (!counter) ? r.temperature : ((r.temperature < obj.low) ? r.temperature : obj.low);
+      totalTemp += r.temperature;
+      counter++;
```

Comment:
Let's say you used some filter method here, what would you do to simplify the calculation of stats?